src/ethereum_provider.js

```diff
@@ -254,7 +254,12 @@ class TrustWeb3Provider extends BaseProvider {
   }
 
   personal_sign(payload) {
-    const message = payload.params[0];
+    var message;
+    if (this.address === payload.params[0]) {
```

Comment:
As far as I know it has nothing to do with the `params` length, only the order.

[this may be caused because we currently allow those params in either order (for bad historical reasons).](https://github.com/MetaMask/metamask-extension/issues/10297#issuecomment-771865389)